### DISCUSSION TITLE
Add Flags to transactions

### DISCUSSION THREAD

**Discussion by Deguello** [actualbudget/releases#72] [Feature ideas]
It's often helpful to be able to mark transactions to see at a glance whether action needs to be taken or where the transaction is in one's workflow. The ability to set a flag status, typically with a color, and, ideally, with a user-configurable label would be very useful for accomplishing this.

I'm sure there are as many ways to utilize this as there are users, but some of the ways I use something like this:

- Flag autopay transactions a given color. I know this is going to happen, well, automatically, and I don't need to do anything with those transactions except wait.
- Flag payments I need to make manually a given color. If I don't want to autopay something for some reason, this flag alerts me that I need to make this payment.
- Flag transactions I expect to be reimbursed. Prompts me to follow up.
- Flag deposits or credits so they stand out more.
- And so on.

Obviously, this is typically done with a flag field and indicator in the register. I think it would also be useful to have the **_option_** to change the background of the entire transaction to the color of the flag for better visibility of the flag status.

**Comment by purpledogeatsgreenpoop** [comment]
This is really the only reason I cannot move to ActualBudget right now.  Because of the way my employer reimburses expenses I rely on flags to indicate if an expense has been reimbursed.  Without flags it's just too painful to migrate.

**Reply by MiriamNZ** [reply]
You can use #reimburse in the memo field.

**Reply by purpledogeatsgreenpoop** [reply]
Agreed; flags are just more convenient ;)

**Reply by purpledogeatsgreenpoop** [reply]
I've been doing this for a couple of months now and its absolutely more cumbersome.  Don't get me wrong, it works, however because of the amount of reimbursed items managed monthly I would change from one color to another to verify everything was correct.  Could just be me set in my old ways too.

**Comment by MiriamNZ** [comment]
I used flags. I set a red flag for anything that needed my attention, and removed the flag when I was done.  Scheduled transactions where the date, amount or notes might change; a transaction I couldn't get right as I entered it  needed to check which categories).

I used a green flag for a "soft reconcile" on the phone. I was glad to drop that when reconciling arrived.
Fuel purchases, which were made in one month and paid in the following month: I set a colour to tie them all together (the actual colour  was irrelevant, it  just had to be different form last month).

I found that I could not remember the meaning of more than two colours. But my memory is getting old now.

Tags work for this, just a bit more effort;  I would use flags if they were there. 
But I would put this low on the priorities for dev attention.
